core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -335,6 +336,47 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  /**
+   * Copy all data from an InputStream to an OutputStream upto maxSize and
+   * closes the input stream if closeStreams is true and all data is read.
+   */
+  def copyStreamUpTo(
+      in: InputStream,
+      maxSize: Long,
+      closeStreams: Boolean = false): (Boolean, InputStream) = {
```

Comment:
I don't think you need to expose `closeStreams`, just have it always be true.  At least I can't think of a reason why it should default to `false` (`copyStream` is used more places ...)